src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
```

Comment:
wouldn't this cause invalid parts of the configuration to be "silently" ignored?